### ISSUE TITLE
Conflicting type for first argument of utoa with stdlib.h declaration

### ISSUE THREAD

**Issue by jlbirccyn** [arduino/Arduino#4002]
Hello

In `hardware/arduino/sam/cores/arduino/itoa.h` utoa is declared as follow:

```
extern char* utoa( unsigned long value, char *string, int radix ) ;
```

in a gcc 4.9.3 for arm distribution, it conflicts with the declaration of the built-in utoa:

```
char *	_EXFUN(__utoa,(unsigned, char *, int));
```

(_EXFUN is defined as follow `#define	_EXFUN(name, proto)		__cdecl name proto
`)

**Comment by cmaglie** [comment]
Is this specific for arm-gcc 4.9.3 or for other versions as well?
Can you provide a test sketch?

**Comment by jlbirccyn** [comment]
Seems to be with 4.9.3. I do not have the error with 4.9.2. No need to provide a sketch. Go in `hardware/arduino/sam/cores/arduino` and type `arm-none-eabi-gcc -c -o test.o -I. WString.cpp` (or whatever name your compiler have, provided it is 4.9.3). You can get a 4.9.3 from https://launchpad.net/gcc-arm-embedded

**Comment by jsanchezgarcia** [comment]
any updates on this?

**Comment by neojou** [comment]
I have the same problem when trying to upgrade toolchain from 4.8.3 to 5.2.1.

**Comment by patrickelectric** [comment]
Same error here :disappointed:

**Comment by jlbirccyn** [comment]
Just sent a pull request to fix the problem

**Comment by cmaglie** [comment]
Should be fixed with #5480.

Please reopen, or furhter comment, if not.

**Comment by clovisf** [comment]
I have tested (**and confirm it works**) this fix with Arduino IDE Nightly Build 1.6.13 on windows 10 using a third-party board with ATSAMD21G18A, after burning the latest Arduino Zero bootloader to it using ATMEL Studio 7.0
